### PR TITLE
Enhanced RDoc for ::new and ::create

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: lib/tempfile.rb
Comment (olleolleolle) [inline lib/tempfile.rb:92]: Is it "based on"?

```suggestion
  # returns a new \Tempfile object based on that file.
```

File: lib/tempfile.rb
Comment (olleolleolle) [inline lib/tempfile.rb:343]: Perhaps here, too "based on"?

```suggestion
# returns a new \File object based on that file.
```

**Review by olleolleolle — COMMENTED** [review]
This is a big step up! Looks good to me!

File: lib/tempfile.rb
Comment (peterzhu2118) [inline lib/tempfile.rb:111]: I'm not sure if this is true. There's a finalizer defined to remove the tempfile when this object is swept by the garbage collector. E.g.

```ruby
irb(main):010:0> f = Tempfile.new
=> #<File:/var/folders/bf/7gzcjlld0ynbrnj0clyc_jgh0000gn/T/20220506-63138-9qcomk>
irb(main):011:0> path = f.path
=> "/var/folders/bf/7gzcjlld0ynbrnj0clyc_jgh0000gn/T/20220506-63138-9qcomk"
irb(main):012:0> File.exist?(path)
=> true
irb(main):013:0> f = nil
=> nil
irb(main):014:0> GC.start
=> nil
irb(main):015:0> File.exist?(path)
=> false
```

File: lib/tempfile.rb
Comment (BurdetteLamar) [reply lib/tempfile.rb]: Fixed.

File: lib/tempfile.rb
Comment (BurdetteLamar) [reply lib/tempfile.rb]: Fixed.

**Comment by BurdetteLamar** [comment]
> This is a big step up! Looks good to me!

Thanks, @olleolleolle.  Now all I need is a review by a committer.

File: lib/tempfile.rb
Comment (BurdetteLamar) [reply lib/tempfile.rb:111]: That's sort of what the previous doc said, I think.  What should we say here?

Comment (peterzhu2118) [reply lib/tempfile.rb:111]: I don't think it said anything about this for the documentation for `Tempfile.new`.

It's certainly true for `Tempfile.create`:

> If no block is given, ... the created file is not removed automatically.

I think we can say something like:

```
The file is removed when the object has died and reclaimed by the garbage collector.
```

Comment (BurdetteLamar) [reply lib/tempfile.rb]: Fixed.

**Comment by BurdetteLamar** [comment]
@peterzhu2118, the re-request review thingy is unresponsive, but I think this is ready for you.

**Review by peterzhu2118 — APPROVED** [review]
Looks good!

**Comment by BurdetteLamar** [comment]
@peterzhu2118, I can't merge here;  can you?

**Comment by peterzhu2118** [comment]
I can't either. But I think @olleolleolle has permissions to merge.

**Review by olleolleolle — APPROVED** [review]
👍

**Comment by olleolleolle** [comment]
I went ahead and merged these changes.

**Pull request merged**
